waltz-ng/client/survey/components/dropdown/survey-dropdown-editor.js

```diff
@@ -62,13 +74,38 @@ function controller() {
     };
 
     vm.updateValue = (entryId, data) => {
-        const entry = _.find(vm.entries, e => e.id === entryId);
+        //entry id is likely to be undefined and cannot be relied upon (when we have new entries)
+        const entry = _.find(vm.entries, e => e.value === data.oldVal);
         entry.value = data.newVal;
         notifyChanges();
     };
 
     vm.removeEntry = (entry) => {
-        _.remove(vm.entries, e => e.id === entry.id);
+        _.remove(vm.entries, e => e.value === entry.value);
```

Comment:
do we want mutation vs. `vm.entries = _.reject(vm.entries, e => e.value === entry.value_);` ?